File: src/controller/gateway_controller.py
from fastapi import APIRouter, HTTPException
from service.gateway_service import get_transformed_user, get_transformed_users
from model.gateway_user_model import GatewayUser, GatewayUserList
from model.error_model import ErrorResponse

router = APIRouter()

@router.get("/user/{user_id}", response_model=GatewayUser, responses={404: {"model": ErrorResponse}})
async def user(user_id: int):
    return await get_transformed_user(user_id)

@router.get("/users", response_model=GatewayUserList)
async def users():
    return await get_transformed_users()

File: src/main/java/com/example/demogateway/controller/GatewayController.java
package com.example.demogateway.controller;

import com.example.demogateway.model.GatewayUser;
import com.example.demogateway.model.GatewayUserList;
import com.example.demogateway.service.GatewayService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/gateway")
public class GatewayController {

    @Autowired
    private GatewayService gatewayService;

    @GetMapping("/user/{id}")
    public ResponseEntity<GatewayUser> user(@PathVariable("id") Integer id) {
        GatewayUser u = gatewayService.getTransformedUser(id);
        return new ResponseEntity<>(u, HttpStatus.OK);
    }

    @GetMapping("/users")
    public ResponseEntity<GatewayUserList> users() {
        GatewayUserList list = gatewayService.getTransformedUsers();
        return new ResponseEntity<>(list, HttpStatus.OK);
    }
}
